### PR TITLE
feat: option to disable trailing block

### PR DESCRIPTION
**Pull request by Kashifwahaj**
```js
//Creates a new editor instance.
  const editor = useCreateBlockNote({
    trailingBlock: true,
  });

  // Renders the editor instance using a React component.
  return <BlockNoteView editor={editor} />;
```
  
  So user need to pass `trailingBlock` boolean option to enable trailing Node.

/claim #631

### CODE REVIEW

File: packages/core/src/editor/BlockNoteEditor.ts [hunk 1]
```diff
@@ -180,6 +182,8 @@ export class BlockNoteEditor<
 
   public readonly uploadFile: ((file: File) => Promise<string>) | undefined;
 
+  public trailingBlock?: boolean = false;
```
Comment (YousefED) [inline packages/core/src/editor/BlockNoteEditor.ts:185]: Can we remove this member, and just pass the option to `getBlockNoteExtensions` on line 249, similar to the other options?

File: packages/core/src/editor/BlockNoteSchema.ts [hunk 2]
```diff
@@ -66,6 +66,8 @@ export class BlockNoteSchema<
      * A list of custom Styles that should be available in the editor.
      */
     styleSpecs?: SSpecs;
+
+    trailingNode?: boolean;
```
Comment (YousefED) [inline packages/core/src/editor/BlockNoteSchema.ts:70]: why is this added here?

**Review by YousefED — COMMENTED** [review]
- can you make the default `true`? (inline with current behaviour)
- can you also update the documentation where the other editor options are explained?

**Comment by Kashifwahaj** [comment]
Got it, I will make the default value `true` and update the documentation also , -I am on it.

File: packages/core/src/editor/BlockNoteEditor.ts (hunk 1 above)
Comment (Kashifwahaj) [reply packages/core/src/editor/BlockNoteEditor.ts:185]: Yes, it's better and resulted in less code. Thanks! 👍

File: packages/core/src/editor/BlockNoteSchema.ts (hunk 2 above)
Comment (Kashifwahaj) [reply packages/core/src/editor/BlockNoteSchema.ts:70]: It was necessary for the 'typescript' warning, but with 'getBlockNoteExtensions', it's no longer needed, so I removed it in the new commits.

**Comment by Kashifwahaj** [comment]
@YousefED 

- [x] Now, the default behavior is `true`. Therefore, if a user explicitly passes `trailingBlock: false`, it will disable the automatic creation of a trailing new block.
- [x] Updated documentation in `/docs/editor-basics/setup#usecreateblocknote-hook` to include the explanation of the new option.

Screenshot 2024-04-09 at 7 28 23 AM


Please let me know if any change is needed.

Thanks

**Comment by Kashifwahaj** [comment]
@YousefED Hi, Please Let me know if any improvement or changes required for this PR.

**Comment by YousefED** [comment]
Looks good, awesome!

**Comment by hampak** [comment]
Hello, @YousefED 

Is #679 going to be released in version 0.12.5?

Thank you for your hard work :)

**Comment by hampak** [comment]
Thank you for this in the latest release :D
